homeassistant/components/sensor/serial.py

```diff
@@ -0,0 +1,78 @@
+"""
+Support for reading data from a serial port.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.serial/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_PORT
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pyserial-asyncio==0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Serial Sensor"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORT): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the Serial sensor platform."""
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_PORT)
+
+    async_add_devices([SerialSensor(name, port)], True)
+
+
+class SerialSensor(Entity):
+    """Representation of a Serial sensor."""
+
+    def __init__(self, name, port):
+        """Initialize the Serial sensor."""
+        self._name = name
+        self._state = None
+        self._port = port
+        self._serial_loop_task = None
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Handle when an entity is about to be added to Home Assistant."""
+        self._serial_loop_task = self.hass.loop.create_task(
```

Comment:
You should first read on EVENT_HOMEASSISTANT_START and stop the task on EVENT_HOMEASSISTANT_STOP with `self._serial_loop_task.cancel()`

Reply:
Thanks, that make sense.